### ISSUE TITLE
Alignment rules missing for rescue/ensure 

### ISSUE THREAD

**Issue by lumeet** [bbatsov/rubocop#2060]
Currently there seems to be no cop to complain about the following.

```ruby
begin
  a
      rescue
        b
            ensure
              c
end
```

And if we start with:
```ruby
begin
a
rescue
b
ensure
c
end
```

...and auto-correct, `Style/IndentationWidth` (alone) makes it to look like this (by default):
```ruby
begin
  a
  rescue
    b
  ensure
    c
end
```

**Comment by bbatsov** [comment]
PRs welcome! :-)

**Comment by lumeet** [comment]
Ok, I'll try to submit one soon as I happened to do some groundwork already.  :) I guess it could be a new `Lint` cop that aligns the `rescue` and `ensure` keywords with the corresponding `end` without introducing any configs. `EndAlignment` and `DefEndAlignment` already do pretty good job to get that `end` aligned correctly. @bbatsov, @jonas054 Any comments at this point?

**Comment by bbatsov** [comment]
Sounds like a plan to me. I've nothing further to add at this point.
